### PR TITLE
Feat/set in discord false controller

### PR DESCRIPTION
**Pull request by RitikJaiswal75**
Create a controller to consume previously created services and set in-discord of a user to false

After a discussion with ankush over the standup call on 15 June 2023 at 6:00 am IST we can skip 5xx tests for this PR.

### CODE REVIEW

File: routes/users.js [hunk 1]
```diff
@@ -9,6 +9,7 @@
 const { getUserBadges } = require("../controllers/badges");
 const checkIsVerifiedDiscord = require("../middlewares/verifydiscord");
 
+router.post("/update-in-discord", authenticate, authorizeRoles([SUPERUSER]), users.setInDiscordScript);
```
[the remark this answers is not in the document]
Comment (harshith-venkatesh) [reply routes/users.js:12]: Could this be looked into please

[the remark this answers is not in the document]
Comment (RitikJaiswal75) [reply routes/users.js:12]: We do not have a rate limiter yet, and the API is limited to only super user so I am not sure about how to solve this

[the remark this answers is not in the document]
Comment (RitikJaiswal75) [reply routes/users.js:12]: https://github.com/Real-Dev-Squad/website-backend/issues/1073

**Review by harshith-venkatesh — APPROVED** [review]
Approving it based on the discussion over the call where @RitikJaiswal75 confirmed 200 or 500 responses for the `setInDiscordScript` .

File: controllers/users.js [hunk 2]
```diff
@@ -558,6 +559,15 @@ const nonVerifiedDiscordUsers = async (req, res) => {
   return res.json(data);
 };
 
+const setInDiscordScript = async (req, res) => {
+  try {
+    await setInDiscordFalseScript();
```
Comment (yesyash) [inline controllers/users.js:564]: - `setInDiscordFalseScript` is this a script?

File: controllers/users.js [hunk 3]
```diff
@@ -558,6 +559,15 @@ const nonVerifiedDiscordUsers = async (req, res) => {
...
+    return res.json({ message: "Successfully added the in_discord field to false for all users" });
```
Comment (yesyash) [inline controllers/users.js:565]: - Should we store this in a constant, than hardcoding the string?

File: test/integration/users.test.js [hunk 4]
```diff
@@ -1161,4 +1161,21 @@ describe("Users", function () {
         });
     });
   });
+
+  describe("POST /update-in-discord", function () {
+    it("it returns proper response", function (done) {
```
Comment (yesyash) [inline test/integration/users.test.js:1166]: - What does "proper response" mean?

**Comment by yesyash** [comment]
> Create a controller to consume previously created services and set in-discord of a user to false
> 
> After a discussion with ankush over the standup call on 15 June 2023 at 6:00 am IST we can skip 5xx tests for this PR.

What is the reason to skip 5xx for tests? 
Please be more descriptive this helps other reviewers.

File: controllers/users.js (hunk 3 above)
Comment (RitikJaiswal75) [reply controllers/users.js:565]: I do not feel so, because this API is just a 1-time thing, creating a constant didn't make sense.

**Comment by RitikJaiswal75** [comment]
> What is the reason to skip 5xx for tests?
Please be more descriptive this helps other reviewers.

The reason to skip 5xx tests is that 5xx only occurs when firebase fails to work, and I was not able to emulate that thus I asked Ankush and he told I cand send the PR and get it merged and send the pr for 5xx errors in future.

File: test/integration/users.test.js (hunk 4 above)
Comment (RitikJaiswal75) [reply test/integration/users.test.js:1166]: It means a not failing, or the mentioned response. 
For a one-time script do we need to focus on the test title as well?

File: controllers/users.js (hunk 2 above)
Comment (RitikJaiswal75) [reply controllers/users.js:564]: it is a function written in a services file which works as a script, it gets all the users in our db and sets a role of in_discord false to them.

**Review by vinayak-trivedi — APPROVED** [review]
LGTM

**Pull request merged**